documentation/USAGE.rst

```diff
@@ -44,6 +44,50 @@ Python can be obtained from the :cite:`PythonReference`.
 As you gain experience, you may want to browse through the
 :ref:`FlagsAndEnvironmentVariables` that affect :term:`FiPy`.
 
+-------
+Logging
+-------
+
+Diagnostic information about a :term:`FiPy` run can be obtained using the
+:mod:`logging` module.  For example, at the beginning of your script, you
+can add::
+
+    >>> import logging
+    >>> log = logging.getLogger("fipy")
+    >>> console = logging.StreamHandler()
+    >>> console.setLevel(logging.INFO)
+    >>> log.addHandler(console)
+
+in order to see informational messages in the terminal.  To have more
+verbose debugging information save to a file::
+
+    >>> logfile = logging.FileHandler(filename="fipy.log")
+    >>> logfile.setLevel(logging.DEBUG)
+    >>> log.addHandler(logfile)
+
+    >>> log.setLevel(logging.DEBUG)
```

Comment:
Is it necessary to set the level twice?

Reply:
It's a debatable point. One establishes what logging messages are captured by a particular handler; the other establishes what messages get generated at all.